x/avs/keeper/keeper.go

```diff
@@ -44,47 +46,83 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 }
 
 func (k Keeper) AVSInfoUpdate(ctx sdk.Context, params *AVSRegisterOrDeregisterParams) error {
+	avsInfo, _ := k.GetAVSInfo(ctx, params.AvsAddress)
+
+	action := params.Action
+
+	if action == RegisterAction && avsInfo != nil {
+		return errorsmod.Wrap(types.ErrAlreadyRegistered, fmt.Sprintf("the error input arg is:%s", params.AvsAddress))
+	}
+
+	avs := &types.AVSInfo{
+		Name:               params.AvsName,
+		AvsAddress:         params.AvsAddress,
+		SlashAddr:          params.SlashContractAddr,
+		AvsOwnerAddress:    params.AvsOwnerAddress,
+		AssetId:            params.AssetID,
+		AvsUnbondingEpochs: uint32(params.MinimumDelegation),
+		MinimumDelegation:  sdk.NewIntFromUint64(params.UnbondingEpochs),
+		AvsEpoch:           nil,
+		OperatorAddress:    nil,
+	}
+	if action == RegisterAction && avsInfo == nil {
+		return k.SetAVSInfo(ctx, avs)
+	}
+
+	if avsInfo == nil {
+		return errorsmod.Wrap(types.ErrUnregisterNonExistent, fmt.Sprintf("the error input arg is:%s", avsInfo))
```

Comment:
avsInfo == nil so this line has no meaning print:   fmt.Sprintf("the error input arg is:%s", avsInfo)

Reply:
unregistration must be done with data available